File: 2894-divisible-and-non-divisible-sums-difference/2894-divisible-and-non-divisible-sums-difference.cpp
class Solution {
public:
    int differenceOfSums(int n, int m) {
        int c = 0;
        int d = 0;
        for (int i = 1; i <= n; i++)
        {
            if (i % m == 0)
            {
                d += i;
            }
            else
            {
                c += i;
            }
        }
        return c - d;
    }
};

File: 2894-divisible-and-non-divisible-sums-difference/2894-divisible-and-non-divisible-sums-difference.rs
impl Solution {
    pub fn difference_of_sums(n: i32, m: i32) -> i32 {
        let mut d = 0;
        for i in 1..n+1{
            if (i % m == 0){
                d -= i;
            }
            else{d += i;}
        }
        d
    }
}